src/replication.c

```diff
@@ -712,6 +712,19 @@ void syncCommand(client *c) {
     /* ignore SYNC if already slave or in monitor mode */
     if (c->flags & CLIENT_SLAVE) return;
 
+    /* Check if this is a failover request to a replica with the same replid and
+     * become a master if so */
+    if (server.masterhost && server.cached_master && (c->argc > 3) &&
```

Comment:
Shouldn't this be server.master instead of cached_master? We only want to accept a failover request if we are currently connected to the master.

Reply:
I don't 100% recall the reason for this. I think that the the issue is the initial primary disconnects all replicas before issuing a PSYNC command to the targeted replica. If the replica processes the primary disconnection before it receives the PSYNC command then server.master will be NULL and failover will fail. server.cached_master will still be valid in this case and allow the failover to complete.